src/main/java/net/logstash/logback/pattern/EnhancedPropertyConverter.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright 2013-2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package net.logstash.logback.pattern;
+
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import ch.qos.logback.classic.pattern.ClassicConverter;
+import ch.qos.logback.classic.pattern.PropertyConverter;
+import ch.qos.logback.classic.spi.ILoggingEvent;
+import ch.qos.logback.classic.spi.LoggerContextVO;
+
+/**
+ * Variation of the legacy Logback {@link PropertyConverter} with the option to specify a
+ * default value to use when the property does not exist instead of returning {@code null}
+ * as does the original Logback implementation.
+ * 
+ * <p>The default value is optional and can be specified using the <code>:-</code> operator as
+ * in Bash shell. For example, assuming the property "foo" is not defined, <code>%property{foo:-bar}</code>
+ * will return <code>bar</code>.
+ * If no optional value is declared, the converter returns an empty string instead of {@code null}
+ * if the property is not defined.
+ * 
+ * <p>The property resolution mechanism is the same as the Logback implementation. The property is
+ * first looked up in the context associated with the logging event. If not found, the property is
+ * searched in the System environment.
+ * 
+ * 
+ * @author brenuart
+ */
+public class EnhancedPropertyConverter extends ClassicConverter {
+    /**
+     * Regex pattern used to extract the optional default value from the key name.
+     */
+    private static final Pattern PATTERN = Pattern.compile("(.+):-(.*)");
```

Comment:
might want to use `(.+?):-(.*)` here, to ensure the property name is before the first `:-`, in case `:-` appears later in the value.